### PR TITLE
feat: ctx request orginator

### PR DESCRIPTION
**Pull request by ig-ra**
tag request orginator via context allowing db layer to know from where request is coming

### CODE REVIEW

File: internal/backend/auth/authhttpmiddleware/authmiddleware.go [hunk 1]
```diff
@@ -36,6 +37,7 @@ type Deps struct {
 func newTokensMiddleware(next http.Handler, tokens authtokens.Tokens) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		ctx := r.Context()
+		ctx = cctx.WithRequestOrinator(ctx, cctx.Middleware)
```
Comment (efiShtain) [inline internal/backend/auth/authhttpmiddleware/authmiddleware.go]: typo

File: internal/backend/db/dbgorm/ownership.go [hunk 2]
```diff
@@ -10,6 +10,7 @@ import (
 
 	"go.autokitteh.dev/autokitteh/internal/backend/auth/authcontext"
 	"go.autokitteh.dev/autokitteh/internal/backend/db/dbgorm/scheme"
+	cctx "go.autokitteh.dev/autokitteh/internal/context"
```
Comment (efiShtain) [inline internal/backend/db/dbgorm/ownership.go]: not sure cctx is a good convention
it's not really readable

you can go with akcontext or internalContext or something more descriptive and less error prone like ctx/cctx

File: internal/backend/db/dbgorm/ownership.go [hunk 3]
```diff
@@ -185,42 +186,42 @@ func joinUserEntity(db *gorm.DB, entity string, uid string) *gorm.DB {
...
 type UsersOwnershipChecker struct {
 	z *zap.Logger
 }
 
 func (c *UsersOwnershipChecker) EnsureUserAccessToEntitiesWithOwnership(
-	db *gorm.DB, uid string, ids ...sdktypes.UUID,
+	ctx context.Context, db *gorm.DB, uid string, ids ...sdktypes.UUID,
 ) ([]scheme.Ownership, error) {
-	c.z.Debug("isUserEntity", zap.Any("entityIDs", ids), zap.Any("uid", uid))
+	c.z.Debug("isUserEntity", zap.String("component", cctx.RequestOrginator(ctx).String()), zap.Any("entityIDs", ids), zap.Any("uid", uid))
```
Comment (efiShtain) [inline internal/backend/db/dbgorm/ownership.go]: component->requestOrigin

Comment (efiShtain) [reply internal/backend/db/dbgorm/ownership.go]: in multiple places

File: internal/backend/workflows/common.go [hunk 4]
```diff
@@ -137,6 +139,7 @@ func (wf *Workflow) StartSessions(wctx workflow.Context, event sdktypes.Event, s
 }
 
 func (wf *Workflow) CreateEventRecord(ctx context.Context, eventID sdktypes.EventID, state sdktypes.EventState) {
+	ctx = cctx.WithRequestOrinator(ctx, cctx.Workflow)
```
Comment (efiShtain) [inline internal/backend/workflows/common.go]: why do you need it here ?
don't you need to define it only once for the workflow?

File: internal/backend/db/dbgorm/ownership.go (hunk 2 above)
Comment (ig-ra) [reply internal/backend/db/dbgorm/ownership.go]: renamed

File: internal/backend/workflows/common.go (hunk 4 above)
Comment (ig-ra) [reply internal/backend/workflows/common.go]: you are right. fixed

File: internal/backend/sessions/sessions.go [hunk 5]
```diff
@@ -71,6 +76,7 @@ func (s *sessions) List(ctx context.Context, filter sdkservices.ListSessionsFilt
 }
 
 func (s *sessions) Delete(ctx context.Context, sessionID sdktypes.SessionID) error {
+	ctx = akCctx.WithRequestOrginator(ctx, akCctx.SessionWorkflow)
```
Comment (efiShtain) [inline internal/backend/sessions/sessions.go]: akCctx -> akCtx

File: internal/context/context.go [hunk 6]
```diff
@@ -0,0 +1,41 @@
+package context
+
+import (
+	"context"
+)
+
+type ctxKey string
+
+const componentCtxKey = ctxKey("component")
```
Comment (efiShtain) [inline internal/context/context.go]:
```suggestion
const originCtxKey = ctxKey("origin")
```

**Review by efiShtain — APPROVED** [review]
two more typos but fine by me

File: internal/backend/sessions/sessions.go (hunk 5 above)
Comment (ig-ra) [reply internal/backend/sessions/sessions.go]: fixed

File: internal/context/context.go [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+	ScheduleWorkflow
+	Middleware
+	Unknown
+)
+
+func (c RequestOrginatorType) String() string {
+	if c >= Unknown {
+		return "unknown"
+	}
+	return [...]string{"dispatcher", "eventsWF", "sessionWF", "schedulerWF", "middleware"}[c]
```
Comment (efiShtain) [inline internal/context/context.go]: use the full name
WF is not a good candidate
sessionsWorkflow
schedulerWorkflow

middleware -> api_call

(middleware has no real meaning)

**Pull request merged**